### PR TITLE
CircleCI 2.0

### PR DESCRIPTION
**Pull request by wyattjoh**
## What does this PR do?

- Adds support for CircleCI 2.0.
- Adds Firefox support for local testing of PR's 🔥 🦊 
- Re-added Safari to integration suite during releases 🍎 
- Added parallel support for CI jobs 🚄 🚋 
- Made CI really, really fast

## How do I test this PR?

- Beauty of this PR, it'll test itself!

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+        command: ./bin/cli setup --defaults
+    - run:
+        name: Run the integration tests
+        command: bash .circleci/e2e.sh
+    - store_test_results:
+        path: /tmp/circleci-test-results
+
+version: 2
+jobs:
+  # npm_dependencies will install the dependancies used by all other steps.
```
Comment (alinhle) [inline .circleci/config.yml]: %s/dependancies/dependencies/g
😉

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+      - test_integration_safari:
+          <<: *filter_deploy
+          requires:
+            - build_assets
+      - deploy:
+          <<: *filter_deploy
+          requires:
+            - lint
+            - test_unit
+            - test_integration_safari
```
Comment (alinhle) [inline .circleci/config.yml]: This used to fail unless BS fixed their issues, does it work?

File: nightwatch.conf.js [hunk 3]
```diff
@@ -49,5 +49,12 @@ module.exports = {
         },
       },
     },
+    'firefox-headless': {
+      desiredCapabilities: {
+        chromeOptions: {
+          args: ['--headless', '--disable-gpu', 'window-size=1600,1200'],
+        },
```
Comment (alinhle) [inline nightwatch.conf.js]: Can you try adding this instead? 
```
'moz:firefoxOptions': {
   args: ['-headless']
 }
```

Obviously `chromeOptions` shouldn't have any effect here..

**Review by alinhle — CHANGES_REQUESTED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (wyattjoh) [reply .circleci/config.yml]: We can remove the requirement, but keep the test there if we want?

Comment (alinhle) [reply .circleci/config.yml]: Yes!

**Review by alinhle — APPROVED** [review]

**Pull request merged**